daemon/server/router/system/system_routes.go

```diff
@@ -230,7 +231,13 @@ func (s *systemRouter) getDiskUsage(ctx context.Context, w http.ResponseWriter,
 			v.LayersSize = systemDiskUsage.Images.TotalSize //nolint: staticcheck,SA1019: v.LayersSize is deprecated: kept to maintain backwards compatibility with API < v1.52, use [ImagesDiskUsage.TotalSize] instead.
 			v.Images = systemDiskUsage.Images.Items         //nolint: staticcheck,SA1019: v.Images is deprecated: kept to maintain backwards compatibility with API < v1.52, use [ImagesDiskUsage.Items] instead.
 		} else if verbose {
-			v.ImageUsage.Items = systemDiskUsage.Images.Items
+			v.ImageUsage.Items = slices.Collect(func(yield func(image.Summary) bool) {
+				for _, i := range systemDiskUsage.Images.Items {
+					if !yield(*i) {
+						return
+					}
+				}
+			})
```

Comment:
Looks like this is needed, because we didn't change the backend type accordingly, right?

Now that the backend type is identical to the type in the API, perhaps we should switch back, but at least we could change it to be a non-pointer as well